lib/WebService/GrowthBook/FeatureRepository.pm

```diff
@@ -0,0 +1,84 @@
+package WebService::GrowthBook::FeatureRepository;
+use strict;
+use warnings;
+no indirect;
+use Scalar::Util qw(blessed);
+use Object::Pad;
+use HTTP::Tiny;
+use Log::Any qw($log);
+use Syntax::Keyword::Try;
+use JSON::MaybeUTF8 qw(decode_json_utf8);
+use WebService::GrowthBook::InMemoryFeatureCache;
+
+## VERSION
+
+class WebService::GrowthBook::FeatureRepository {
+    field $http :param :writer //= HTTP::Tiny->new();
+    field $cache :param //= WebService::GrowthBook::InMemoryFeatureCache->singleton();
+    method set_cache($new_cache){
+        die "Invalid cache object $new_cache" unless blessed($new_cache) && $new_cache->isa('WebService::GrowthBook::AbstractFeatureCache');
+        $cache = $new_cache;
+    }
+    
+    method clear_cache(){
+        $cache->clear();
+    }
+    method load_features($api_host, $client_key, $ttl = 60) {
+        my $key = $api_host . '::' . $client_key;
```

Comment:
Would we have multiple cache keys we would have to worry about ? Since we should be using a single api key within the application. Also Should we use the api_key as part  of the cache key ?

Reply:
we cached all features together, api_host + client_key will map to a single result. so it is ok to combine api_host and client key. In the future, if we have requirements to cache single feature, then we can improve it at that time.
I will change client_key to md5 string

Reply:
if all the features are stored under a single key. do we need to dynamically generate a key ? Could we not use a static key name ?